### PR TITLE
throw error if git version is less than 2

### PR DESCRIPTION
**Pull request by kostiakoval**
Fix https://bugs.swift.org/browse/SR-966.
Require git with version >= 2

### CODE REVIEW

File: Sources/Get/Error.swift [hunk 1]
```diff
@@ -38,6 +39,8 @@ extension Error: CustomStringConvertible {
             return "No version tag found in (\(package)) package. Add a version tag with \"git tag\" command. Example: \"git tag 0.1.0\""
         case NoManifest(let clonePath, let version):
             return "The package at `\(clonePath)' has no Package.swift for the specific version: \(version)"
+        case ObsoleteGitVersion:
+            return "Git 2.0 or higer is required. Please update git and retry."
```
Comment (aciidb0mb3r) [inline Sources/Get/Error.swift]: Spelling mistake: higer -> higher

File: Sources/Utility/Git.swift [hunk 2]
```diff
@@ -51,4 +51,18 @@ public class Git {
...
+    public class var version: String! {
+        return try? popen([Git.tool, "version"])
+    }
+
+    public class var majorVersionNumber: Int? {
+        let prefix = "git version"
+        var version = self.version
+        if version.hasPrefix(prefix) {
+            let prefixRange = version.startIndex...version.startIndex.advanced(by: prefix.characters.count)
+            version.removeSubrange(prefixRange)
```
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: Can just do `version = version[version.startIndex.advance(by: prefix.characters.count)..<version.endIndex]`

File: Sources/Utility/Git.swift [hunk 3]
```diff
@@ -51,4 +51,18 @@ public class Git {
...
+        }
+        return Int(String(version.characters.first!))
```
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: Avoid bang?

File: Sources/Get/get().swift [hunk 4]
```diff
@@ -18,6 +18,10 @@ import Utility
  - Returns: The modules that this manifest requires building
 */
 public func get(manifest: Manifest, manifestParser: (path: String, url: String) throws -> Manifest) throws -> [Package] {
+    if Git.majorVersionNumber < 2 {
```
Comment (aciidb0mb3r) [inline Sources/Get/get().swift]: maybe 2 can be a constant something like kMinGitVersionRequired

Comment (kostiakoval) [inline Sources/Get/get().swift]: maybe just adding description would be enough? 
We don's use this constant in other places

```swift
// Git 2.0 or higher is required
if Git.majorVersionNumber < 2 {
  throw Error.ObsoleteGitVersion
}
```
vs
```swift
let kMinGitVersionRequired = 2
if Git.majorVersionNumber < kMinGitVersionRequired {
  throw Error.ObsoleteGitVersion
}
```

File: Sources/Utility/Git.swift (hunk 2 above)
Comment (kostiakoval) [inline Sources/Utility/Git.swift]: String is not a sequence, so I can't :(

**Comment by mxcl** [comment]
Specifically what I wanted here was *only* to check the git version in the event of another git error being thrown.

File: Sources/Utility/Git.swift (hunk 2 above)
Comment (kostiakoval) [inline Sources/Utility/Git.swift]: I can do this, but in that case there are to many expressions in 1 line, I think
```swift
version = String(version.characters[version.characters.startIndex.advanced(by: prefix.characters.count)..<version.characters.endIndex])
```

**Comment by kostiakoval** [comment]
Got it :+1:.  Is there a convenient way to downgrade git?

File: Sources/Utility/Git.swift (hunk 2 above)
Comment (aciidb0mb3r) [inline Sources/Utility/Git.swift]: yeah that'll be worse 😅

**Comment by mxcl** [comment]
Homebrew can be coaxed into installing old versions.

**Comment by kostiakoval** [comment]
All calls to git `system(Git.tool, ...)` and `popen([Git.tool, ...)` could fail. Functions that uses them should be marked with `throws`.

When the Git error occurs and we catch it. We can: 
- Handle it in place and terminate program, `exit(1)`
- Propagate error up, so that `Multitool` would handle it. `handleError(msg: Any, ...)` method   
    With this solution methods that invokes Git should be marked as `throws` and that would require refactoring quite a lot of code.

I think I prefer solution nr.1 We should handle error in place and take necessary actions. System can't work without git, so we should terminate it. 

Do you agree or have any other solutions proposal ?

**Comment by mxcl** [comment]
Yes, seems like the right choice. Unless properties become capable of throwing at some point.

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by kostiakoval** [comment]
Tests are passing. Any more feedback on this one. If no, then I'm going to merge it

**Comment by mxcl** [comment]
Next time please squash commits like "revert necessary changes" away first.

**Comment by kostiakoval** [comment]
I'm sorry. I will do that next time

**Comment by mxcl** [comment]
np, every project has its unwritten rules.

**Pull request merged**